### PR TITLE
Longhaul Test Documentation: Refactor docs and add longhaul doc

### PR DESCRIPTION
**Pull request by and-rewsmith**
Adding a longhaul document. But also took the time to consolidate all our other test documentation into one folder.

I can go in when I have more time to flesh this doc out more. But I just wanted to put the images from my presentation with a tad of context.

### CODE REVIEW

**Comment by damonbarry** [comment]
> Example:
> https://github.com/Azure/iotedge/blob/d3641ff6f2a8e2bffd8050869ad2eb649ae16677/test/doc/longhaul.md

Nit: The commit ID in this URL is specific to your PR, so I _think_ it will go away once the PR is merged. If this URL becomes part of your commit message it will be a broken link. I suggest removing the reference.

File: test/doc/longhaul.md
Comment (damonbarry) [inline test/doc/longhaul.md:18]: I think of a "dummy module" as a module that doesn't do anything. Is that what you're describing?

Comment (and-rewsmith) [reply test/doc/longhaul.md:18]: They are dumb in the sense that the modules are doing things only for testing. It is better to call them test modules I think.

Comment (damonbarry) [reply test/doc/longhaul.md]: Does LonghaulTest_General.png need to change too?

Comment (and-rewsmith) [reply test/doc/longhaul.md]: I think with the wording it is fine, but I can change it if you want.

Comment (damonbarry) [reply test/doc/longhaul.md]: What I mean is, the markdown file calls them "test modules", but LonghaulTest_General.png still has "dummy module 1" and "dummy module 2". Your call.

**Review by damonbarry — APPROVED** [review]

**Pull request merged**
